iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -4,1310 +4,1043 @@
+            //TODO stubbing this method because we plan on removing it from the device client later.
+            Message message = null;
+            return Task.FromResult(message);
         }
 
-        protected override void Dispose(bool disposing)
+        public override async Task EnableMethodsAsync(CancellationToken cancellationToken)
         {
-            try
-            {
-                if (Logging.IsEnabled)
-                    Logging.Enter(this, $"{nameof(DefaultDelegatingHandler)}.Disposed={_disposed}; disposing={disposing}", $"{nameof(MqttTransportHandler)}.{nameof(Dispose)}");
-
-                if (!_disposed)
-                {
-                    base.Dispose(disposing);
-                    if (disposing)
-                    {
-                        if (TryStop())
-                        {
-                            CleanUpAsync().GetAwaiter().GetResult();
-                        }
-
-                        _disconnectAwaitersCancellationSource?.Dispose();
-                        _disconnectAwaitersCancellationSource = null;
-
-                        _receivingSemaphore?.Dispose();
-                        _receivingSemaphore = null;
-
-                        _deviceReceiveMessageSemaphore?.Dispose();
-                        _deviceReceiveMessageSemaphore = null;
-
-                        if (_channel is IDisposable disposableChannel)
-                        {
-                            disposableChannel.Dispose();
-                            _channel = null;
-                        }
-                    }
+            await SubscribeAsync(DirectMethodsRequestTopic, cancellationToken);
+        }
 
-                    // the _disposed flag is inherited from the base class DefaultDelegatingHandler and is finally set to null there.
-                }
-            }
-            finally
-            {
-                if (Logging.IsEnabled)
-                    Logging.Exit(this, $"{nameof(DefaultDelegatingHandler)}.Disposed={_disposed}; disposing={disposing}", $"{nameof(MqttTransportHandler)}.{nameof(Dispose)}");
-            }
+        public override async Task DisableMethodsAsync(CancellationToken cancellationToken)
+        {
+            await UnsubscribeAsync(DirectMethodsRequestTopic, cancellationToken);
         }
 
-        public override async Task CloseAsync(CancellationToken cancellationToken)
+        public override async Task SendMethodResponseAsync(DirectMethodResponse methodResponse, CancellationToken cancellationToken)
         {
-            try
-            {
-                if (Logging.IsEnabled)
-                    Logging.Enter(this, "", $"{nameof(MqttTransportHandler)}.{nameof(CloseAsync)}");
+            var topic = DirectMethodsResponseTopicFormat.FormatInvariant(methodResponse.Status, methodResponse.RequestId);
 
-                cancellationToken.ThrowIfCancellationRequested();
+            MqttApplicationMessage mqttMessage = new MqttApplicationMessageBuilder()
```

Comment:
Can we cache the MqttApplicationMessageBuilder so we don't call new on it so much?

Reply:
Might be able to include some of the calls below to that cached instance, like topic and QoS.

Reply:
Oh, maybe not because then each additional call to it would modify the builder. Hmm.

Reply:
The only point of commonality between every message is the QoS, so I don't see a lot of value in trying to cache this builder